compare/compare.go

```diff
@@ -15,30 +16,89 @@ type CompareParams struct {
 	DisallowExtraFields  bool
 }
 
-// Compare ...
-func Compare(expected, actual interface{}, params CompareParams) []error {
+type PureCompareType int
+
+const (
+	Pure PureCompareType = iota
+	Regex
+)
+
+// Deep compares arrays or maps
+func Deep(expected, actual interface{}, params CompareParams) []error {
 	return compareBranch("$", expected, actual, &params)
 }
 
-func compareBranch(path string, expected, actual interface{}, params *CompareParams) []error {
+// Plain compares values as plain text
+// It can be compared several ways:
+// - Pure values: should be equal
+// - Regex: try to compile 'expected' as regex and match 'actual' with it
+func Plain(expected, actual interface{}, cmpType PureCompareType) []error {
+	return plain("$", expected, actual, cmpType)
+}
+
+func plain(path string, expected, actual interface{}, cmpType PureCompareType) []error {
+
+	var errors []error
+
 	expectedType := getType(expected)
 	actualType := getType(actual)
-	var errors []error
 
 	// compare types
 	if expectedType != actualType {
 		errors = append(errors, makeError(path, "types do not match", expectedType, actualType))
 		return errors
 	}
 
-	// compare scalars
-	if isScalarType(actualType) && !params.IgnoreValues && !isScalarsEqual(expected, actual) {
-		errors = append(errors, makeError(path, "values do not match", expected, actual))
+	switch cmpType {
```

Comment:
Okay. The plan was to not bring mess into Compare function. Failed. :)

Please do the following:
- let's leave Deep (I'm ok with your renaming) func internals untouched (there is plan to replace it with some external lib)
- let's create another file in compare package: compare_plain.go, put new function Plain into it.
- Implement Plain from scratch as a new string comparison function (with optional regexp mode)
- call from Check() either compare.Deep or compare.Plain according to response mime-type

Reply:
I get your point. You want to split Plain and Deep compare completely. Ok, I don't mind.

But I want to note - if we do this, it will be more difficult to bring regex in Deep() func. What if we want to have it at leaf nodes of the tree? But I'm not sure if anyone will need it in the future.
How do you think?

Reply:
👍for thinking in advance.

What if we would support some custom syntax that can be used in place of regular strings, like:
```
    response:
        200: >
            {
                  "token": "$matchRegexp(/^\w{16}$/)"
            }
```

Then in your case that would be almost the same way:
```
    response:
        200: $matchRegexp(/^Some string \w{16} pattern defined as an expression$/)
```

`$func()` syntax is already used in fixtures, so support of it in test definition fits to the overall tool concept.

Also, that would result to that flag IsRegexp is not necessary, as regexp are available all the time. 
And in that way you don't need to split Compare function, just improve the `isScalarsEqual` function.

How does this sound to you?

Reply:
Yeah, I thought about a similar way initially, but I didn't guess about the $func.
Sounds good, I like it. Let's do it this way.